bind.go

```diff
@@ -138,6 +138,9 @@ func (b *DefaultBinder) bindData(destination interface{}, data map[string][]stri
 	// You are better off binding to struct but there are user who want this map feature. Source of data for these cases are:
 	// params,query,header,form as these sources produce string values, most of the time slice of strings, actually.
 	if typ.Kind() == reflect.Map && typ.Key().Kind() == reflect.String {
+		if val.IsNil() {
```

Comment:
could you move this block of checking for nil and creating the map to just before `for k, v := range data {`. In that case we do not create that map for cases where we would return before setting map elements

and these 2 broken tests check for `map[string]int(nil)` and `map[string][]int(nil)`